src/page-modules/assistant/details/index.tsx

```diff
@@ -70,12 +71,13 @@ export function AssistantDetails({ tripPattern }: AssistantDetailsProps) {
           <TripSection
             key={index}
             isFirst={index === 0}
-            isLast={index === tripPattern.legs.length}
+            isLast={index === tripPattern.legs.length - 1}
             leg={leg}
             interchangeDetails={getInterchangeDetails(
               tripPattern.legs,
               leg.interchangeTo?.toServiceJourney.id,
             )}
+            legWaitDetails={getLegWaitDetails(leg, tripPattern.legs[index + 1])}
```

Comment:
This kind of still feels odd for me. Both interchange and this. I don't have any better options for now, but feels kind of fragmented domain logic wise 🤔

Reply:
We might want to handle this server side?

Reply:
Yes! Good point. If this is just data transformation I think it should happen in the service layer.